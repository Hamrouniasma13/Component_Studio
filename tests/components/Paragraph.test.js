import React from 'react';
import renderer from 'react-test-renderer';

import Paragraph from '../../components/Paragraph';

const loremLine =
  'Sint minim consequat velit irure consectetur eu non consectetur aute deserunt labore velit aliqua mollit.';

describe('Component: <Paragraph />', () => {
  it('render <p/> with simple text', () => {
    const tree = renderer.create(<Paragraph content={loremLine} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('render <div />', () => {
    const tree = renderer
      .create(<Paragraph tag="div" content={loremLine} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('render react component', () => {
    const tree = renderer
      .create(
        <Paragraph
          content={
            <>
              <h1>{loremLine}</h1>
            </>
          }
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
