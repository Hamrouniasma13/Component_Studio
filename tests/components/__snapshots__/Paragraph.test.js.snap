// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Paragraph /> render <div /> 1`] = `
.emotion-0 {
  font-size: 16px;
  line-height: 1.5;
  font-family: Source Sans Pro,Arial,Helvetica,sans-serif;
  color: black;
}

.emotion-0 p > a {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #494df4;
  font-weight: bold;
}

@media only screen and (min-width :768px) {
  .emotion-0 {
    font-size: 18px;
    line-height: 1.67;
  }
}

<div
  className="emotion-0"
>
  Sint minim consequat velit irure consectetur eu non consectetur aute deserunt labore velit aliqua mollit.
</div>
`;

exports[`Component: <Paragraph /> render <p/> with simple text 1`] = `
.emotion-0 {
  font-size: 16px;
  line-height: 1.5;
  font-family: Source Sans Pro,Arial,Helvetica,sans-serif;
  color: black;
}

.emotion-0 p > a {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #494df4;
  font-weight: bold;
}

@media only screen and (min-width :768px) {
  .emotion-0 {
    font-size: 18px;
    line-height: 1.67;
  }
}

<p
  className="emotion-0"
>
  Sint minim consequat velit irure consectetur eu non consectetur aute deserunt labore velit aliqua mollit.
</p>
`;

exports[`Component: <Paragraph /> render react component 1`] = `
.emotion-0 {
  font-size: 16px;
  line-height: 1.5;
  font-family: Source Sans Pro,Arial,Helvetica,sans-serif;
  color: black;
}

.emotion-0 p > a {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #494df4;
  font-weight: bold;
}

@media only screen and (min-width :768px) {
  .emotion-0 {
    font-size: 18px;
    line-height: 1.67;
  }
}

<div
  className="emotion-0"
>
  <h1>
    Sint minim consequat velit irure consectetur eu non consectetur aute deserunt labore velit aliqua mollit.
  </h1>
</div>
`;
